src/executor.ts

```diff
@@ -140,7 +142,12 @@ export class Executor {
       if (config.command === undefined) {
         execution = new NoCommandScriptExecution(config, this, this._logger);
       } else if (config.service) {
-        execution = new ServiceScriptExecution(config, this, this._logger);
+        execution = new ServiceScriptExecution(
+          config,
+          this,
+          this._logger,
+          this._abort.promise
```

Comment:
Why aren't you using an AbortSignal?

Reply:
Because it's only supported in Node 15+, and we still support Node 14 until it goes out of LTS next year. Simpler than having a polyfill since it's so trivial anyway.